### PR TITLE
Add retrieve os image component for Azure

### PR DESCRIPTION
**Pull request by ChiChiHeHeZhou**
## What this PR does / why we need it

Add Retrieve OS Image component for AzureClusterSettingsStep.tsx
## Which issue(s) this PR fixes

Fixes: #5055

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/stores/Azure.store.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ const initialState = {
         AZURE_ENVIRONMENT: AzureCloud.PUBLIC,
         REGION: '',
         SSH_PUBLIC_KEY: '',
+        IMAGE_INFO: {},
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/stores/Azure.store.tsx:20]: I would rename this to OS_IMAGE for clarity and simplicity

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-basic/AzureManagementClusterBasic.constants.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ export const enum AZURE_FIELDS {
     AZURE_ENVIRONMENT = 'AZURE_ENVIRONMENT',
     REGION = 'REGION',
     SSH_PUBLIC_KEY = 'SSH_PUBLIC_KEY',
+    IMAGE_INFO = 'IMAGE_INFO',
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-basic/AzureManagementClusterBasic.constants.tsx:14]: Here as well

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx [hunk 3]
```diff
@@ -103,6 +113,21 @@ export function AzureClusterSettingsStep(props: Partial<StepProps>) {
...
+    const onOsImageSelected = (imageName: string) => {
+        images.some((image) => {
+            if (image.name === imageName) {
+                setImageParameters(image);
+            }
+        });
+    };
+
+    useEffect(() => {
+        AzureService.getAzureOsImages().then((data) => {
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:125]: Should handle error state by providing a notification "Unable to retrieve a list of OS images: " and include the body of the error text.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx [hunk 4]
```diff
@@ -127,6 +152,16 @@ export function AzureClusterSettingsStep(props: Partial<StepProps>) {
                             selectedInstanceId={selectedInstanceTypeId}
                         />
                     </div>
+                    <div cds-layout="col:12">
+                        <RetrieveOSImages
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:156]: Sorry I think I missed this in your previous PR for the component name. I would rename the component to OsImageSelect

**Review by miclettej — COMMENTED** [review]
Very good, just a few hopefully minor edits to make. Thanks!

File: cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/stores/Azure.store.tsx (hunk 1 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/stores/Azure.store.tsx:20]: Done

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-basic/AzureManagementClusterBasic.constants.tsx (hunk 2 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-basic/AzureManagementClusterBasic.constants.tsx:14]: Done

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx (hunk 3 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:125]: Done

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx (hunk 4 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:156]: Done

**Review by miclettej — APPROVED** [review]

**Pull request merged**